Sources/ApolloCodegenLib/Frontend/CompilationResult.swift

```diff
@@ -173,6 +181,27 @@ public class CompilationResult: JavaScriptObject {
 
     lazy var directives: [Directive]? = self["directives"]
 
+    lazy var isDeferred: IsDeferred = {
```

Comment:
This is duplicated for both `InlineFragment` and `FragmentSpread`, I'm not sure if there is a better way to do it; they're both just simple `JavaScriptObject` types.

Reply:
```swift
protocol Deferrable {
  var directives: [Directive]? { get }
}

fileprivate extension Deferrable where Self: JavaScriptObject {
  func getIsDeferred() -> IsDeferred {
    ...
  }
}

public class InlineFragment: Deferrable {
  ...
  lazy var isDeferred: IsDeferred = getIsDeferred()
}
```

Wish we could easily namespace this `Deferrable` protocol in `CompilationResult`. Until we break `CompilationResult` into it's own package, we might want to call the protocol `CompilationResult_Deferrable`

Reply:
The protocol and extension can have the `fileprivate` scope which is effectively the same as putting it into it's own package for this particular case.